File: board-room/src/main/java/net/stzups/board/room/server/websocket/protocol/client/messages/ClientMessageDeleteDocument.java
package net.stzups.board.room.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.room.server.websocket.protocol.client.ClientMessage;
import net.stzups.board.room.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageDeleteDocument extends ClientMessage {
    private long id;

    public ClientMessageDeleteDocument(ByteBuf byteBuf) {
        super(ClientMessageType.DELETE_DOCUMENT);
        this.id = byteBuf.readLong();
    }

    public long id() {
        return id;
    }
}

File: board-frontend/src/scripts/protocol/client/messages/ClientMessageDeleteDocument.js
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";

export default class ClientMessageDeleteDocument extends ClientMessage {
    constructor(document) {
        super(ClientMessageType.DELETE_DOCUMENT);
        this.id = document.id;
    }

    getBufferSize() {
        return super.getBufferSize() + 8;
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeBigInt64(this.id);
    }
}